src/config/headers.rs

```diff
@@ -10,6 +11,12 @@ pub struct Headers {
     /// activated. The `max-age` value is the least of the values received from
     /// upstream services. @default `false`.
     pub cache_control: Option<bool>,
+
+    #[serde(default, skip_serializing_if = "is_default")]
+    /// `headers` are key-value pairs included in every server
+    /// response. Useful for setting headers like `Access-Control-Allow-Origin`
+    /// for cross-origin requests or additional headers for downstream services.
+    pub custom: Vec<KeyValue>,
 }
```

Comment:
Can we create a new type `Custom` here so that the merging logic could be moved here.